# Organize Netflix movie actor rating data

## 1. Analysis Objective

The purpose of this data analysis is to organize actor rating data for films and TV works across different genres, such as comedy, action, and sci-fi. By calculating the average IMDb rating of the works each actor has appeared in within each genre, this analysis aims to identify actors associated with high-rated works in each genre.

The purpose of this practical project is to practice data cleaning and organization, so as to obtain a dataset that can be used for further analysis.

## 2. Introduction

The original dataset records all Netflix movies and TV shows available in the United States as of July 2022. The dataset contains two data tables: `titles.csv` and `credits.csv`.

`titles.csv` contains information related to movies and TV shows, including the title ID, title, type, description, genres, IMDb rating, and other attributes. IMDb is an international online platform for film and TV ratings. `credits.csv` contains information on more than 70,000 directors and actors who appeared in Netflix titles, including their names, title IDs, character names, and roles in the production, such as director or actor.

The meanings of each column in `titles.csv` are as follows:

- `id`: The ID of the film or TV title.  
- `title`: The title of the film or TV show.  
- `type`: The type of title, either a TV show or a movie.  
- `description`: A brief description.  
- `release_year`: The release year.  
- `age_certification`: The age rating.  
- `runtime`: The duration of each TV episode or movie.  
- `genres`: A list of genres.  
- `production_countries`: A list of production countries.  
- `seasons`: The number of seasons, if the title is a TV show.  
- `imdb_id`: The IMDb ID.  
- `imdb_score`: The IMDb rating.  
- `imdb_votes`: The number of IMDb votes.  
- `tmdb_popularity`: The TMDB popularity score.  
- `tmdb_score`: The TMDB rating.  
  
The meanings of each column in `credits.csv` are as follows:  
  
- `person_id`: The ID of the cast or crew member.  
- `id`: The ID of the film or TV title they participated in.  
- `name`: The name of the person.  
- `character`: The name of the character.  
- `role`: The role type, either actor or director.  


## Load Data

Import the library needed for data analysis, then use pandas read_csv() function to load the raw data file `titles.csv` and `credits.csv`.

In [1]:
import pandas as pd
import numpy as np

In [2]:
original_titles = pd.read_csv("./titles.csv")
original_titles

,id,title,type,description,release_year,age_certification,runtime,genres,production_countries,seasons,imdb_id,imdb_score,imdb_votes,tmdb_popularity,tmdb_score
0,ts300399,Five Came Back: The Reference Films,SHOW,This collection includes 12 World War II-era p...,1945,TV-MA,51,['documentation'],['US'],1.0,NaN,NaN,NaN,0.600,NaN
1,tm84618,Taxi Driver,MOVIE,A mentally unstable Vietnam War veteran works ...,1976,R,114,"['drama', 'crime']",['US'],NaN,tt0075314,8.2,808582.0,40.965,8.179
2,tm154986,Deliverance,MOVIE,Intent on seeing the Cahulawassee River before...,1972,R,109,"['drama', 'action', 'thriller', 'european']",['US'],NaN,tt0068473,7.7,107673.0,10.010,7.300
3,tm127384,Monty Python and the Holy Grail,MOVIE,"King Arthur, accompanied by his squire, recrui...",1975,PG,91,"['fantasy', 'action', 'comedy']",['GB'],NaN,tt0071853,8.2,534486.0,15.461,7.811
4,tm120801,The Dirty Dozen,MOVIE,12 American military prisoners in World War II...,1967,NaN,150,"['war', 'action']","['GB', 'US']",NaN,tt0061578,7.7,72662.0,20.398,7.600
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
5845,tm1014599,Fine Wine,MOVIE,A beautiful love story that can happen between...,2021,NaN,100,"['romance', 'drama']",['NG'],NaN,tt13857480,6.8,45.0,1.466,NaN
5846,tm898842,C/O Kaadhal,MOVIE,A heart warming film that explores the concept...,2021,NaN,134,['drama'],[],NaN,tt11803618,7.7,348.0,NaN,NaN
5847,tm1059008,Lokillo,MOVIE,A controversial TV host and comedian who has b...,2021,NaN,90,['comedy'],['CO'],NaN,tt14585902,3.8,68.0,26.005,6.300
5848,tm1035612,Dad Stop Embarrassing Me - The Afterparty,MOVIE,"Jamie Foxx, David Alan Grier and more from the...",2021,PG-13,37,[],['US'],NaN,NaN,NaN,NaN,1.296,10.000


In [3]:
original_credits = pd.read_csv("./credits.csv")
original_credits

,person_id,id,name,character,role
0,3748,tm84618,Robert De Niro,Travis Bickle,ACTOR
1,14658,tm84618,Jodie Foster,Iris Steensma,ACTOR
2,7064,tm84618,Albert Brooks,Tom,ACTOR
3,3739,tm84618,Harvey Keitel,Matthew 'Sport' Higgins,ACTOR
4,48933,tm84618,Cybill Shepherd,Betsy,ACTOR
...,...,...,...,...,...
77796,736339,tm1059008,Adelaida Buscato,María Paz,ACTOR
77797,399499,tm1059008,Luz Stella Luengas,Karen Bayona,ACTOR
77798,373198,tm1059008,Inés Prieto,Fanny,ACTOR
77799,378132,tm1059008,Isabel Gaona,Cacica,ACTOR


## Data evaluation and cleaning

### Back up original data

In [4]:
cleaned_titles = original_titles.copy()
cleaned_credits = original_credits.copy()

### Evaluation and cleaning of `titles.csv`

#### Data tidiness

Each column represents a variable and each row represents an observation. However, this project aims to organize the average IMDb scores of works across different genres and actors. Therefore, the lists in the `genres` column need to be split into separate rows. However, since there is no additional requirement related to countries, the lists in the `production_countries` column do not need to be split.

Convert the `genres` column from string format into actual Python lists。

In [5]:
cleaned_titles['genres'] = cleaned_titles['genres'].apply(lambda s: eval(s)) 

After converting the column into lists, we can use the DataFrame explode() method to split the list values in that column into separate rows, which makes the data easier to clean and arrange later.

In [6]:
cleaned_titles = cleaned_titles.explode("genres")
cleaned_titles

,id,title,type,description,release_year,age_certification,runtime,genres,production_countries,seasons,imdb_id,imdb_score,imdb_votes,tmdb_popularity,tmdb_score
0,ts300399,Five Came Back: The Reference Films,SHOW,This collection includes 12 World War II-era p...,1945,TV-MA,51,documentation,['US'],1.0,NaN,NaN,NaN,0.600,NaN
1,tm84618,Taxi Driver,MOVIE,A mentally unstable Vietnam War veteran works ...,1976,R,114,drama,['US'],NaN,tt0075314,8.2,808582.0,40.965,8.179
1,tm84618,Taxi Driver,MOVIE,A mentally unstable Vietnam War veteran works ...,1976,R,114,crime,['US'],NaN,tt0075314,8.2,808582.0,40.965,8.179
2,tm154986,Deliverance,MOVIE,Intent on seeing the Cahulawassee River before...,1972,R,109,drama,['US'],NaN,tt0068473,7.7,107673.0,10.010,7.300
2,tm154986,Deliverance,MOVIE,Intent on seeing the Cahulawassee River before...,1972,R,109,action,['US'],NaN,tt0068473,7.7,107673.0,10.010,7.300
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
5847,tm1059008,Lokillo,MOVIE,A controversial TV host and comedian who has b...,2021,NaN,90,comedy,['CO'],NaN,tt14585902,3.8,68.0,26.005,6.300
5848,tm1035612,Dad Stop Embarrassing Me - The Afterparty,MOVIE,"Jamie Foxx, David Alan Grier and more from the...",2021,PG-13,37,NaN,['US'],NaN,NaN,NaN,NaN,1.296,10.000
5849,ts271048,Mighty Little Bheem: Kite Festival,SHOW,"With winter behind them, Bheem and his townspe...",2021,NaN,7,family,[],1.0,tt13711094,7.8,18.0,2.289,10.000
5849,ts271048,Mighty Little Bheem: Kite Festival,SHOW,"With winter behind them, Bheem and his townspe...",2021,NaN,7,animation,[],1.0,tt13711094,7.8,18.0,2.289,10.000


#### Data quality

In [7]:
cleaned_titles.info()

<class 'pandas.DataFrame'>
Index: 15147 entries, 0 to 5849
Data columns (total 15 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   id                    15147 non-null  str    
 1   title                 15146 non-null  str    
 2   type                  15147 non-null  str    
 3   description           15125 non-null  str    
 4   release_year          15147 non-null  int64  
 5   age_certification     9298 non-null   str    
 6   runtime               15147 non-null  int64  
 7   genres                15088 non-null  str    
 8   production_countries  15147 non-null  str    
 9   seasons               5923 non-null   float64
 10  imdb_id               14525 non-null  str    
 11  imdb_score            14399 non-null  float64
 12  imdb_votes            14375 non-null  float64
 13  tmdb_popularity       14995 non-null  float64
 14  tmdb_score            14614 non-null  float64
dtypes: float64(5), int64(2), str(8)
memo

The dataset contains 15147 rows and 15 columns. Columns of `title`, `description`, `age_certification`, `genres`, `seasons`, `imdb_id`, `imdb_score`, `imdb_votes`, `tmdb_popularity`, and `tmdb_score` contain missing values.

##### Missing data

Based on the project purpose, missing values in the `genres` and `imdb_score` columns would directly affect the analysis. Therefore, rows with missing values in these columns should be removed.

Remove rows with missing `genres` or `imdb_score` values, and then check the remaining number of missing values in these columns.

In [8]:
cleaned_titles.dropna(subset = "genres", inplace = True)
cleaned_titles.dropna(subset = "imdb_score", inplace = True)
print(cleaned_titles['genres'].isna().sum())
print(cleaned_titles['imdb_score'].isna().sum())

0
0


##### Duplicated data

In [9]:
cleaned_titles.duplicated().sum()

np.int64(0)

The dataset does not contain duplicated rows.

In [10]:
cleaned_titles.drop_duplicates(subset=['id', 'genres'], inplace = True)

After exploding the genres column, each combination of `id` and `genres` should be unique. Therefore, duplicated rows based on these two columns were removed.

##### Inconsistent data

It is impossible to predict all forms of inconsistent data. However, inconsistent values in the `genres` column may affect the analysis results, so they should be converted into a consistent format before analysis. Inconsistent data in other columns can be ignored in this case because they are less relevant to the project objective.

In [11]:
cleaned_titles["genres"].value_counts()

genres
drama            2827
comedy           2218
thriller         1180
action           1109
romance           955
crime             909
documentation     859
family            652
animation         630
fantasy           619
scifi             560
european          428
horror            366
history           254
music             244
reality           219
sport             170
war               155
western            39
Name: count, dtype: int64

No inconsistent values were found in the `genres` column.

##### Invalid or erroneous data assessment

In [12]:
cleaned_titles.describe()

,release_year,runtime,seasons,imdb_score,imdb_votes,tmdb_popularity,tmdb_score
count,14393.000000,14393.000000,5655.000000,14393.000000,1.437100e+04,14268.000000,13975.000000
mean,2015.994164,78.326270,2.424934,6.534802,2.836162e+04,30.008057,6.885143
std,7.156617,40.555015,2.867856,1.157513,1.041849e+05,99.359539,1.112448
min,1954.000000,0.000000,1.000000,1.500000,5.000000e+00,0.600000,1.000000
25%,2015.000000,44.000000,1.000000,5.800000,7.080000e+02,3.911750,6.200000
50%,2018.000000,87.000000,1.000000,6.600000,3.166000e+03,9.666000,6.985000
75%,2020.000000,106.000000,3.000000,7.400000,1.432100e+04,22.766000,7.600000
max,2022.000000,225.000000,42.000000,9.500000,2.294231e+06,2274.044000,10.000000


The `imdb_score` column does not contain any values below 0 or above 10. Therefore, no invalid IMDb scores were found. No other unrealistic or unreasonable values were found in the dataset.

### Evaluation and cleaning `credits.csv`

#### Data tidiness

In [13]:
cleaned_credits.sample(10)

,person_id,id,name,character,role
7274,11064,tm74780,Rob Morrow,Dr. Hollins,ACTOR
28902,916475,tm412550,Kate Middleton,Mary,ACTOR
47669,86174,tm548602,Scott Aukerman,NaN,DIRECTOR
2031,17364,tm43808,Larry Kaplan,Uncle Ted,ACTOR
71132,5834,tm1134006,Laurence Fishburne,Narrator (voice),ACTOR
13190,1202,tm60292,Bill Tangradi,Alan B. Golacinski,ACTOR
76155,140749,tm1030241,Sudheesh,NaN,ACTOR
24833,281652,tm300713,Hiroomi Tosaka,Amamiya Hiroto,ACTOR
5757,15030,tm45454,Matthew Glave,Secret Service Agent Driver,ACTOR
36871,50623,tm299504,Secun de la Rosa,Carlos,ACTOR


From a random sample of 10 rows, each column represents a variable, each row represents an observation, and each cell contains a single value. Therefore, the dataset is tidy.

#### Data quality

In [14]:
cleaned_credits.info()

<class 'pandas.DataFrame'>
RangeIndex: 77801 entries, 0 to 77800
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype
---  ------     --------------  -----
 0   person_id  77801 non-null  int64
 1   id         77801 non-null  str  
 2   name       77801 non-null  str  
 3   character  68029 non-null  str  
 4   role       77801 non-null  str  
dtypes: int64(1), str(4)
memory usage: 3.0 MB


The dataset contains 77801 rows and 5 columns. Most columns have complete values, while only the `character` column contains missing values.

The role column contains two consistent categories: ACTOR and DIRECTOR. Since this project focuses on actors’ average IMDb scores by genre, DIRECTOR records are not relevant to the analysis objective. Therefore, only records with role = ACTOR will be kept for the final analysis.

In [15]:
cleaned_credits = cleaned_credits[cleaned_credits["role"] == "ACTOR"]
cleaned_credits["role"].value_counts()

role
ACTOR    73251
Name: count, dtype: int64

All DIRECTOR records were removed, and only ACTOR records were kept.

##### Missing data

Missing values were found only in the `character` column. Since this column is not essential for calculating actors’ average IMDb scores by genre, these missing values were kept and no rows will be removed.

##### Duplicated data

In [16]:
cleaned_credits.duplicated().sum()

np.int64(0)

The dataset does not contain duplicated rows.

Duplicates based on `person_id` and `id` will be checked because the same actor should only be counted once for the same title. Otherwise, the same IMDb score may be counted multiple times and affect the average rating calculation.

In [17]:
cleaned_credits.duplicated(subset=['id', 'person_id']).sum()

np.int64(88)

Since duplicated combinations of `id` and `person_id` were found, these duplicates were removed to ensure that each actor was counted only once for each title.

In [18]:
cleaned_credits.drop_duplicates(subset = ["id", "person_id"], inplace = True)
cleaned_credits.duplicated(subset=["id", "person_id"]).sum()

np.int64(0)

Remove rows with duplicate `id` and `person_id` values, and then check whether duplicated combinations still exist.

##### Inconsistent data

It is impossible to predict all forms of inconsistent data. However, inconsistent values in the `role` column may affect the analysis results, so they should be converted into a consistent format before analysis. Inconsistent data in other columns can be ignored in this case because they are less relevant to the project objective.

In [19]:
cleaned_credits["role"].value_counts()

role
ACTOR    73163
Name: count, dtype: int64

No inconsistent data were found in the `role` column.

##### Invalid or erroneous data

In [20]:
cleaned_credits.describe()

,person_id
count,7.316300e+04
mean,5.345023e+05
std,6.473876e+05
min,7.000000e+00
25%,4.401200e+04
50%,1.975820e+05
75%,9.009495e+05
max,2.462818e+06


The `credits.csv` dataset mainly contains categorical and identifier columns. No values in the `person_id` were less than or equal to 0. Therefore, no obvious invalid or erroneous data were found in the key columns used for this project.

## Save cleaned data

Save the cleaned datasets in the same folder.

In [21]:
cleaned_titles.to_csv("cleaned_titles.csv", index = False)

In [22]:
cleaned_credits.to_csv("cleaned_credits.csv", index = False)

Read the saved files to verify that they were exported successfully.

In [23]:
pd.read_csv("./cleaned_titles.csv")

,id,title,type,description,release_year,age_certification,runtime,genres,production_countries,seasons,imdb_id,imdb_score,imdb_votes,tmdb_popularity,tmdb_score
0,tm84618,Taxi Driver,MOVIE,A mentally unstable Vietnam War veteran works ...,1976,R,114,drama,['US'],NaN,tt0075314,8.2,808582.0,40.965,8.179
1,tm84618,Taxi Driver,MOVIE,A mentally unstable Vietnam War veteran works ...,1976,R,114,crime,['US'],NaN,tt0075314,8.2,808582.0,40.965,8.179
2,tm154986,Deliverance,MOVIE,Intent on seeing the Cahulawassee River before...,1972,R,109,drama,['US'],NaN,tt0068473,7.7,107673.0,10.010,7.300
3,tm154986,Deliverance,MOVIE,Intent on seeing the Cahulawassee River before...,1972,R,109,action,['US'],NaN,tt0068473,7.7,107673.0,10.010,7.300
4,tm154986,Deliverance,MOVIE,Intent on seeing the Cahulawassee River before...,1972,R,109,thriller,['US'],NaN,tt0068473,7.7,107673.0,10.010,7.300
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
14388,tm898842,C/O Kaadhal,MOVIE,A heart warming film that explores the concept...,2021,NaN,134,drama,[],NaN,tt11803618,7.7,348.0,NaN,NaN
14389,tm1059008,Lokillo,MOVIE,A controversial TV host and comedian who has b...,2021,NaN,90,comedy,['CO'],NaN,tt14585902,3.8,68.0,26.005,6.300
14390,ts271048,Mighty Little Bheem: Kite Festival,SHOW,"With winter behind them, Bheem and his townspe...",2021,NaN,7,family,[],1.0,tt13711094,7.8,18.0,2.289,10.000
14391,ts271048,Mighty Little Bheem: Kite Festival,SHOW,"With winter behind them, Bheem and his townspe...",2021,NaN,7,animation,[],1.0,tt13711094,7.8,18.0,2.289,10.000


In [24]:
pd.read_csv("./cleaned_credits.csv")

,person_id,id,name,character,role
0,3748,tm84618,Robert De Niro,Travis Bickle,ACTOR
1,14658,tm84618,Jodie Foster,Iris Steensma,ACTOR
2,7064,tm84618,Albert Brooks,Tom,ACTOR
3,3739,tm84618,Harvey Keitel,Matthew 'Sport' Higgins,ACTOR
4,48933,tm84618,Cybill Shepherd,Betsy,ACTOR
...,...,...,...,...,...
73158,368473,tm1059008,Aída Morales,Maritza,ACTOR
73159,736339,tm1059008,Adelaida Buscato,María Paz,ACTOR
73160,399499,tm1059008,Luz Stella Luengas,Karen Bayona,ACTOR
73161,373198,tm1059008,Inés Prieto,Fanny,ACTOR


## Merge DataFrames

Both DataFrames have the column `id`, `cleaned_titles` and `cleaned_credits` should be merged based on the column `id`.

In [25]:
merged_df = pd.merge(cleaned_titles, cleaned_credits, on = "id", how = "inner")
merged_df

,id,title,type,description,release_year,age_certification,runtime,genres,production_countries,seasons,imdb_id,imdb_score,imdb_votes,tmdb_popularity,tmdb_score,person_id,name,character,role
0,tm84618,Taxi Driver,MOVIE,A mentally unstable Vietnam War veteran works ...,1976,R,114,drama,['US'],NaN,tt0075314,8.2,808582.0,40.965,8.179,3748,Robert De Niro,Travis Bickle,ACTOR
1,tm84618,Taxi Driver,MOVIE,A mentally unstable Vietnam War veteran works ...,1976,R,114,drama,['US'],NaN,tt0075314,8.2,808582.0,40.965,8.179,14658,Jodie Foster,Iris Steensma,ACTOR
2,tm84618,Taxi Driver,MOVIE,A mentally unstable Vietnam War veteran works ...,1976,R,114,drama,['US'],NaN,tt0075314,8.2,808582.0,40.965,8.179,7064,Albert Brooks,Tom,ACTOR
3,tm84618,Taxi Driver,MOVIE,A mentally unstable Vietnam War veteran works ...,1976,R,114,drama,['US'],NaN,tt0075314,8.2,808582.0,40.965,8.179,3739,Harvey Keitel,Matthew 'Sport' Higgins,ACTOR
4,tm84618,Taxi Driver,MOVIE,A mentally unstable Vietnam War veteran works ...,1976,R,114,drama,['US'],NaN,tt0075314,8.2,808582.0,40.965,8.179,48933,Cybill Shepherd,Betsy,ACTOR
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
205172,tm1059008,Lokillo,MOVIE,A controversial TV host and comedian who has b...,2021,NaN,90,comedy,['CO'],NaN,tt14585902,3.8,68.0,26.005,6.300,368473,Aída Morales,Maritza,ACTOR
205173,tm1059008,Lokillo,MOVIE,A controversial TV host and comedian who has b...,2021,NaN,90,comedy,['CO'],NaN,tt14585902,3.8,68.0,26.005,6.300,736339,Adelaida Buscato,María Paz,ACTOR
205174,tm1059008,Lokillo,MOVIE,A controversial TV host and comedian who has b...,2021,NaN,90,comedy,['CO'],NaN,tt14585902,3.8,68.0,26.005,6.300,399499,Luz Stella Luengas,Karen Bayona,ACTOR
205175,tm1059008,Lokillo,MOVIE,A controversial TV host and comedian who has b...,2021,NaN,90,comedy,['CO'],NaN,tt14585902,3.8,68.0,26.005,6.300,373198,Inés Prieto,Fanny,ACTOR


## Performing groupby operations on the DataFrame.

Create a pivot table to calculate the average IMDb score for each actor within each genre.

In [26]:
initial_results = pd.pivot_table(merged_df, index = ["genres","person_id"], values = 'imdb_score', aggfunc = np.mean)
initial_results

imdb_score
genres  person_id            
action  45                5.0
        48                5.4
        51                6.4
        53                6.8
        54                5.3
...                       ...
western 2353339           6.9
        2370848           6.1
        2398539           3.8
        2406218           6.0
        2408082           7.3

[168881 rows x 1 columns]

Reset the index to convert `genres` and `person_id` back into regular columns, making the result easier to sort, filter, and analyze.

In [27]:
results_reset_index = initial_results.reset_index()
results_reset_index

,genres,person_id,imdb_score
0,action,45,5.0
1,action,48,5.4
2,action,51,6.4
3,action,53,6.8
4,action,54,5.3
...,...,...,...
168876,western,2353339,6.9
168877,western,2370848,6.1
168878,western,2398539,3.8
168879,western,2406218,6.0


For each genre, find the highest average IMDb score among all actors.

In [28]:
genres_max_scores = results_reset_index.groupby("genres")["imdb_score"].max()
genres_max_scores

genres
action           9.3
animation        9.3
comedy           9.2
crime            9.5
documentation    9.1
drama            9.5
european         8.9
family           9.3
fantasy          9.3
history          9.1
horror           9.0
music            8.8
reality          8.9
romance          9.2
scifi            9.3
sport            9.1
thriller         9.5
war              8.8
western          8.9
Name: imdb_score, dtype: float64

Merge the maximum scores back to the actor-level results to find which actors achieved the highest average IMDb score in each genre.

In [29]:
genres_max_scores_person_id = pd.merge(results_reset_index, genres_max_scores, on = ["genres", "imdb_score"])
genres_max_scores_person_id

,genres,person_id,imdb_score
0,action,1303,9.3
1,action,12790,9.3
2,action,21033,9.3
3,action,86591,9.3
4,action,336830,9.3
...,...,...,...
131,war,826547,8.8
132,western,22311,8.9
133,western,28166,8.9
134,western,28180,8.9


Create a reference table that maps each `person_id` to the corresponding actor `name`, so that actor names can be merged back into the final result. Duplicated rows were removed to avoid repeated matches during the merge. The `person_id` column was also checked to ensure that each actor ID appears only once in this reference table.

In [30]:
person_id_names = cleaned_credits[["person_id", "name"]].drop_duplicates(subset=["person_id"])
person_id_names

,person_id,name
0,3748,Robert De Niro
1,14658,Jodie Foster
2,7064,Albert Brooks
3,3739,Harvey Keitel
4,48933,Cybill Shepherd
...,...,...
77794,369221,Ana María Sánchez
77796,736339,Adelaida Buscato
77797,399499,Luz Stella Luengas
77798,373198,Inés Prieto


Merge the actor names into the maximum-score result table, sort the results by `genres`, reset the index, and remove the old index column.

In [31]:
genres_max_score_names = pd.merge(genres_max_scores_person_id, person_id_names, on = 'person_id').sort_values('genres').reset_index().drop('index', axis = 1)
genres_max_score_names

,genres,person_id,imdb_score,name
0,action,1303,9.3,Jessie Flower
1,action,12790,9.3,Olivia Hack
2,action,21033,9.3,Zach Tyler
3,action,86591,9.3,Cricket Leigh
4,action,336830,9.3,André Sogliuzzo
...,...,...,...,...
131,war,826547,8.8,Yuto Uemura
132,western,28180,8.9,Unsho Ishizuka
133,western,22311,8.9,Koichi Yamadera
134,western,28166,8.9,Megumi Hayashibara


## Save the final results

Save the final result as a CSV file so that it can be used for further analysis or presentation.

In [32]:
genres_max_score_names.to_csv("genres_max_score_names.csv", index=False)

Read the saved file to verify that it was exported successfully.

In [33]:
pd.read_csv("genres_max_score_names.csv")

,genres,person_id,imdb_score,name
0,action,1303,9.3,Jessie Flower
1,action,12790,9.3,Olivia Hack
2,action,21033,9.3,Zach Tyler
3,action,86591,9.3,Cricket Leigh
4,action,336830,9.3,André Sogliuzzo
...,...,...,...,...
131,war,826547,8.8,Yuto Uemura
132,western,28180,8.9,Unsho Ishizuka
133,western,22311,8.9,Koichi Yamadera
134,western,28166,8.9,Megumi Hayashibara
